### ISSUE TITLE
Spreadsheet fails to export to csv when cells has formulas

### ISSUE THREAD

**Issue by rprebollo** [ONLYOFFICE/DesktopEditors#92]
When try to export an active sheet to CSV, cells with formulas ends like "no" content. I am using Linux Mint 18.3. I downloaded OnlyOffice from github's latest release page (4.8 DesktopEditors). I was digging on source code but would need some guidance to get this done by myself.

**Comment by ShockwaveNN** [comment]
@rprebollo Could you attach document which cause this behavior.
And describe in details that do you mean by "cells with formulas ends like "no" content"

**Comment by ShockwaveNN** [comment]
If you mean by this `"Save cells content as formulas instead of calculated values".` - as this option named in LibreOffice - we consider to adding this in future (enchasment 36913 in our private ticket system)
Currently there is no such way to do so.

**Comment by rprebollo** [comment]
@ShockwaveNN I am sorry for the delay and for my english too. I tried to attach the file but am getting "We don't support that file type".
I think i can give you the steps to reproduce. I create an .xlsx with three fields in the first row, they will be csv headers. In second row gave them some values: first string literal, second numeric (formula), third a formula that combines text and numbers.
(Original xslx)

field1 |  field2 | field3
-------- | --------- | ---------
Test field with no formula | =ROW()-1 | =CONCAT(ROW()-1,"_Hello")

The process of building this example lead me to some "findings". When i try to "Save as .." and check the .csv content get strange results for values of the third column, was expecting ¨1_Hello¨ but in place give me "field2" ???????. My first assumption was wrong, it is not with all formulas, field2 gets the correct value, it seems just with formulas that combines text and numbers.
(csv result)

field1,field2,field3
Test field with no formula,1,field2

Off course, this is not actual .xslx but works to show undesired behaviour. Actual .xslx in cells with the "problematic" formulas when exported to csv was getting "no" literal string. ??????
I am currently looking for an alternative to LibreOffice, gets stuck every 5 seconds with my big content cells, OnlyOffice Desktop Editors Spreadsheet does pretty well by i need export to .csv works well too.
I followed your suggestion and remove my post from stackoverflow.
For the record, you gays are making a stirring job here!!!

**Comment by ShockwaveNN** [comment]
@rprebollo I confirm I reproduce your problem in current dev version of DocumentServer if understand your correctly
Simplifyed your case:
1. Type `Hello` to A1
2. Type `=CONCAT(A1, 'World')` to A2
3. Make sure that A2 display HelloWorld
4. Export to csv and csv will look like `Hello`

I created bug 36922 in our private ticket system, but it take some time to appear in DesktopEditors

**Comment by ShockwaveNN** [comment]
About `Actual .xslx in cells with the "problematic" formulas when exported to csv was getting "no" literal string. ??????` - I think you selected incorrect encoding while exporting csv, it's not related to bug with formula as I think

**Comment by rprebollo** [comment]
@ShockwaveNN As for your simplified case exactly: the bug appears when there is another cell involved in formula. There will be some feedback here when bug be fixed???
About "no" literal string, in DE Spreadsheet i can't even see exporting csv encoding options but maybe that is the cause.

**Comment by ShockwaveNN** [comment]
>There will be some feedback here when bug be fixed
I'm pretty sure it will be fixed in next major release of DocumentServer, but I try to forget notifiy in this thread when it will be fixed

> i can't even see exporting csv encoding options
Sorry I mislead you, this option currenly exist only in DocumentServer, not in DesktopEditors, we think about adding it in future

**Comment by rprebollo** [comment]
@ShockwaveNN thanks for all, i will be "**watching**"

**Comment by ShockwaveNN** [comment]
@rprebollo Hi there. Your issue is fixed https://github.com/ONLYOFFICE/core/commit/dec099b19bb5c9f2ced3a5b5dd3bbcf264a4b7a4
It will be aviable in next stable release of DocumentServer (in month or two) and in next major release of DesktopEditors (coming some time after DocumentServer)

**Comment by ShockwaveNN** [comment]
@rprebollo sorry, there is no such instruction

**Comment by rrebollo** [comment]
@ShockwaveNN hi, i just saw the new release, read the "Fixes" section but see none mention of this bug. Could you give me some clarification? When will be ready linux's releases?

**Comment by ShockwaveNN** [comment]
@rrebollo It was fixed in 5.1.0 and you right, there is no such changelog entry for DocumentServer, changelogs for major releases are not 100% detailed since there is a lot of changes.
This bug is fixed and should be ok in 5.1.27

**Comment by rrebollo** [comment]
@ShockwaveNN sorry to bother you, wish could give you something more than my thanks, i am gonna test it. I think you could ckeck this as closed.

**Comment by rrebollo** [comment]
@ShockwaveNN sorry to bother you, wish could give you something more than my thanks, i tested bug in new release and ... i found a new bug, sorry. My original .xslx has multiple sheets, six to be more accurate and when i export to .csv is always exporting the same sheet, no matter what active sheet is, and is always the fifth. Some clue??

**Comment by ShockwaveNN** [comment]
@rprebollo which desktopversion are you using? If latest (5.1.29.350 in About menu) - please attach this file, I cannot reproduce it on latest version

**Comment by rrebollo** [comment]
Yes, latest version, linux mint 18.3 x64.here is the document
[datos.xlsx](https://github.com/ONLYOFFICE/DesktopEditors/files/2049151/datos.xlsx). Please let me know if there is anything i can do.

**Comment by Rita-Bubnova** [comment]
Hello, @rprebollo.
Thank you. I can confirm - this is a bug, issue 37858 in our internal issue tracker.

**Comment by rrebollo** [comment]
@Rita-Bubnova When it will be fixes? Can i delete my document from here?

**Comment by Rita-Bubnova** [comment]
Yeah, you can delete it if you want, it will be fixed but currently we have not ETA.

**Comment by rrebollo** [comment]
@Rita-Bubnova Thanks, maybe you could give me some feedback when issue 37858 is fixed for any release???
